src/parser/parser_verification.js

```diff
@@ -0,0 +1,150 @@
+import { requiredValues } from '../util/requiredValues';
+import { parserUtils } from './parser_utils';
+
+/**
+ * Verify and trigger warnings if a node required value is not set.
+ * @param  {Node} node - The node element.
+ * @param  {Function} emit - Emit function used to trigger Warning event.
+ * @param  {Boolean} [isAdInline] - True if node is contained inside a InLine node.
+ *
+ */
+function verifyRequiredValues(node, emit, isAdInline) {
+  if (!node || !node.nodeName) {
+    return;
+  }
+  if (node.nodeName === 'InLine') {
+    isAdInline = true;
+  }
+  verifyRequiredAttributes(node, emit);
+
+  if (hasSubElements(node)) {
+    verifyRequiredSubElements(node, emit, isAdInline);
+    for (const nodeKey in node.children) {
+      verifyRequiredValues(node.children[nodeKey], emit, isAdInline);
+    }
+  } else if (parserUtils.parseNodeText(node).length === 0) {
+    emitMissingValueWarning({ node }, emit);
+    return;
+  }
+
+  return;
+}
+
+/**
+ * Verify and trigger warnings if node required attributes are not set.
+ * @param  {Node} node - The node element.
+ * @param  {Function} emit - Emit function used to trigger Warning event.
+ */
+function verifyRequiredAttributes(node, emit) {
+  if (!requiredValues[node.nodeName]) {
+    return;
+  }
+  const requiredAttributes = requiredValues[node.nodeName].attributes;
+  for (const attributeKey in requiredAttributes) {
+    const attributeName = requiredAttributes[attributeKey];
+    const attribute = node.getAttribute(attributeName);
+    if (!attribute) {
+      emitMissingValueWarning(
+        { node, name: attributeName, type: 'attribute' },
+        emit
+      );
+    }
+  }
+}
+
+/**
+ * Verify and trigger warnings if node required sub element are not set.
+ * @param  {Node} node - The node element
+ * @param  {Boolean} isAdInline - True if node is contained in a inline
+ * @param  {Function} emit - Emit function used to trigger Warning event.
+ */
+function verifyRequiredSubElements(node, emit, isAdInline) {
+  const required = requiredValues[node.nodeName];
+  if (!required) {
+    return;
+  }
+  const requiredSubElements = required.subElements;
```

Comment:
this could be undefined as well, as not every requiredValues from the config has subElements mentioned

Reply:
Yep! I'll add the check